common/utilities/time/waiting-on.h

```diff
@@ -0,0 +1,134 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <mutex>
+#include <condition_variable>
+#include <thread>
+#include <atomic>
+
+
+namespace utilities {
+namespace time {
+
+// Helper class -- encapsulate a variable of type T that we want to wait on: another thread will set
+// it and signal when we can continue...
+//
+template< class T >
+class waiting_on
+{
+    // We need to be careful with timeouts: if we time out then the local waiting_on can go out of
+    // scope and then the thread cannot set anything or signal anyone! We get around this by using a
+    // shared_ptr & weak_ptr to manage access:
+    //
+public:
+    class wait_state_t
+    {
+        T _value;
+        std::condition_variable _cv;
+
+        friend class waiting_on;
+
+    public:
+        wait_state_t() = default;   // allow default ctor
+        wait_state_t( T const & t )
+            : _value( t )
+        {
+        }
+
+        operator T &() { return _value; }

```

Comment:
we want to enable to change the value without signal?

Reply:
Sure, the set-and-signal function is just a convenience function